apps/annotations/models.py

```diff
@@ -113,28 +113,27 @@ def add_tag(self, tag: Tag, team: Team, added_by: CustomUser = None):
         self.tags.add(tag, through_defaults={"team": team, "user": added_by})
 
     def user_tag_names(self):
-        return {tag["name"] for tag in self.tags_json if not tag["is_system_tag"]}
+        return {tag["name"] for tag in self.prefetched_tags_json if not tag["is_system_tag"]}
 
     def system_tags_names(self):
-        return {(tag["name"], tag["category"]) for tag in self.tags_json if tag["is_system_tag"]}
+        return {(tag["name"], tag["category"]) for tag in self.prefetched_tags_json if tag["is_system_tag"]}
 
     def all_tag_names(self):
-        return [tag["name"] for tag in self.tags_json]
+        return [tag["name"] for tag in self.prefetched_tags_json]
 
     @cached_property
-    def tags_json(self):
-        tagged_items = CustomTaggedItem.objects.filter(
-            content_type__model=self._meta.model_name, content_type__app_label=self._meta.app_label, object_id=self.id
-        ).prefetch_related("tag", "user")
+    def prefetched_tags_json(self):
+        if not hasattr(self, "prefetched_tagged_items"):
```

Comment:
let's add a warning log here to make it more visible if the prefetch hasn't been done.